### PR TITLE
Allow disabling activation offloading streams in full finetune recipe

### PR DESCRIPTION
**Pull request by nathan-az**
#### Context
What is the purpose of this PR? Is it to
- [ ] add a new feature
- [x] fix a bug
- [ ] update tests and/or documentation
- [ ] other (please add here)

Please link to any issues this PR addresses.

https://github.com/pytorch/torchtune/issues/2683

#### Changelog
What are the changes made in this PR?
* Allows configuring `use_streams` in activation offloading

#### Test plan
Please make sure to do each of the following if applicable to your PR. If you're unsure about any one of these just ask and we will happily help. We also have a [contributing page](https://github.com/pytorch/torchtune/blob/main/CONTRIBUTING.md) for some guidance on contributing.

- [ ] run pre-commit hooks and linters (make sure you've first installed via `pre-commit install`)
- [ ] add [unit tests](https://github.com/pytorch/torchtune/tree/main/tests/torchtune) for any new functionality
- [ ] update [docstrings](https://github.com/pytorch/torchtune/tree/main/docs/source) for any new or updated methods or classes
- [ ] run unit tests via `pytest tests`
- [ ] run recipe tests via `pytest tests -m integration_test`
- [ ] manually run any new or modified recipes with sufficient proof of correctness
- [ ] include relevant commands and any other artifacts in this summary (pastes of loss curves, eval results, etc.)

#### UX
If your function changed a public API, please add a dummy example of what the user experience will look like when calling it.
Here is a [docstring example](https://github.com/pytorch/torchtune/blob/6a7951f1cdd0b56a9746ef5935106989415f50e3/torchtune/modules/vision_transformer.py#L285)
and a [tutorial example](https://pytorch.org/torchtune/main/tutorials/qat_finetune.html#applying-qat-to-llama3-models)

- [ ] I did not change any public API
- [ ] I have added an example to docs or docstrings

### CODE REVIEW

**Comment by nathan-az** [comment]
@joecummings @janeyx99 Let me know if this looks like a reasonable stopgap. I'm not certain that it completely solves the issue, but is a reasonable stopgap (if anyone can repro both the instability in the issue, and that this change fixes it. I assume that if the issue is the streams, fixing it may be quite a bit higher effort.

**Comment by janeyx99** [comment]
@nathan-az I think allowing a nonstreams option is good! Your inclination is right that if it is streams that’s the problem, it will be nontrivial to fix.

**Review by ebsmothers — APPROVED** [review]
@nathan-az thanks for opening this! I agree, this looks like a reasonable fix to unblock TP with activation offloading
